usaspending_api/references/v2/views/filter_trees/tas_filter_tree.py

```diff
@@ -0,0 +1,94 @@
+from usaspending_api.common.helpers.agency_logic_helpers import cfo_presentation_order
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from usaspending_api.common.helpers.agency_logic_helpers import agency_from_identifiers
+from usaspending_api.references.v2.views.filter_trees.filter_tree import DEFAULT_CHILDREN, Node, FilterTree
+from usaspending_api.references.models import ToptierAgency
+
+
+class TASFilterTree(FilterTree):
+    def toptier_search(self):
+        agency_id_sets = TreasuryAppropriationAccount.objects.values("fr_entity_code", "agency_id").distinct(
+            "agency_id"
+        )
+        agency_set = {
+            ToptierAgency.objects.filter(
+                toptier_code=agency_from_identifiers(elem["agency_id"], elem["fr_entity_code"])
+            ).first()
+            for elem in agency_id_sets
+        }
+        agency_dictionaries = [self._dictionary_from_agency(agency) for agency in agency_set]
+        cfo_sort_results = cfo_presentation_order(agency_dictionaries)
+        return cfo_sort_results["cfo_agencies"] + cfo_sort_results["other_agencies"]
+
+    def _dictionary_from_agency(self, agency):
+        if agency:
+            return {"toptier_code": agency.toptier_code, "name": agency.name}
+        else:
+            return {"toptier_code": "-1", "name": "failed to find agency"}
```

Comment:
Usually we don't show agencies that are not in our toptier table.  I'm pretty sure we don't want -1 "failed to find agency" in our response (which I'm seeing).